tests/test_cmake_build/embedded.cpp

```diff
@@ -0,0 +1,21 @@
+#include <pybind11/embedded.h>
+namespace py = pybind11;
+
+PYBIND11_ADD_EMBEDDED_MODULE(test_cmake_build)(py::module &m) {
+    m.def("add", [](int i, int j) { return i + j; });
```

Comment:
One positive change of this: it always seemed redundant to me to need to specify "test_cmake_build" twice: once in the macro, and once as a string in the `py::module` constructor.

Could we get rid of the necessity of the arguments from the macro by throwing `(py::module &m)` into the macro?  While I appreciate it being explicit, I feel it's the sort of thing that is *always* going to be specified exactly as is.